tests/suites/test_suite_x509write.function

```diff
@@ -10,27 +10,31 @@
 #include "mbedtls/pk.h"
 #include "mbedtls/psa_util.h"
 
-#if defined(MBEDTLS_RSA_C)
-int mbedtls_rsa_decrypt_func(void *ctx, size_t *olen,
-                             const unsigned char *input, unsigned char *output,
-                             size_t output_max_len)
+#if defined(MBEDTLS_PEM_WRITE_C) && \
+    defined(MBEDTLS_X509_CRT_WRITE_C) && \
+    defined(MBEDTLS_X509_CRT_PARSE_C) && \
+    defined(MBEDTLS_MD_CAN_SHA1) && \
```

Comment:
Note: we have ongoing work to change certain dependency symbols inclduing `MBEDTLS_MD_CAN_SHA1` on the `development` branch, and this will cause a conflict. However the work for this particular symbol hasn't been scheduled yet AFAICT so hopefully this pull request will be merged before then.

Reply:
Fingers crossed 😁

Regarding sample programs: I also noticed that some dependencies went of out sync with the messages provided by some programs, e.g.:

`MBEDTLS_CIPHER_C ` vs. `MBEDTLS_MD_C`:

https://github.com/Mbed-TLS/mbedtls/blob/24a0908eca5255059ba97e78f6f68cd391424e36/programs/cipher/cipher_aead_demo.c#L43-L50

`MBEDTLS_FS_IO` hint missing in `mbedtls_printf`:

https://github.com/Mbed-TLS/mbedtls/blob/24a0908eca5255059ba97e78f6f68cd391424e36/programs/ssl/ssl_server.c#L12-L23

There might be more, those are the ones I remember.

Quick idea coming to mind would be to test for each one alone and have dedicated `printf`'s for each one missing. This would of course add more lines per sample.

Reply:
> I also noticed that some dependencies went of out sync with the messages provided by some programs

Yes, that definitely could use improvement..

> test for each one alone and have dedicated printf's for each one missing

Yes, that would be better.

Both are low-priority concerns though, so I can't guarantee that we would do or review such a change promptly.

Reply:
Ok, good to know you agree, so effort spent wisely :-)

No problem, I will have to look into that when I find the time as well.